pkg/operator/ceph/object/spec.go

```diff
@@ -461,6 +468,20 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) (v1.Container,
 	return container, nil
 }
 
+func (c *clusterConfig) getRGWContainerCMD(args *[]string) []string {
+	if c.isLocalizedReadAffinitySet() {
+		// $hostName in the args comes from the bash script variable with the same name.
+		*args = append(*args, cephconfig.NewFlag("crush location", fmt.Sprintf("%s=%s", "host", "$hostName")))
+		parseArgsForScript(args)
+		// CrushMap HostName is parsed to replace "." with "-". `hostName=\"$(echo $NODE_NAME | tr . -)\"` does that.
+		// $NODE_NAME in the script is the node name passed to the radosgw container as downward API
+		rgwCmd := "hostName=\"$(echo $NODE_NAME | tr . -)\" && radosgw " + strings.Join(*args, " ")
```

Comment:
As I mentioned, I think we need to use `exec radosgw <args>` for this to work properly with signal handling. `exec` stops the `bash` process and replaces it with `radosgw` which is important for the container env.